src/gluonts/ev/metrics.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from gluonts.exceptions import GluonTSUserError
+from .api import (
+    DerivedMetricEvaluator,
+    MetricEvaluator,
+    StandardMetricEvaluator,
+)
+from .aggregations import Mean, Sum
+from .stats import (
+    absolute_error,
+    absolute_label,
+    absolute_percentage_error,
+    absolute_scaled_error,
+    coverage,
+    quantile_loss,
+    scaled_interval_score,
+    squared_error,
+    symmetric_absolute_percentage_error,
+)
+
+
+def assert_axis_is_None_or_one(axis: Optional[int], metric_name: str) -> None:
```

Comment:
I don't like this. What is special about axis 1?

Reply:
I agree that axis 1 shouldn't be anything special. Just to clarify what lead to this: In my head, a univariate time series looks like this:

                       time stamps
                       t1 t2 t3 t4 ...
time series I:  o  o  o  o
time series II: o  o  o  o
...

This check is supposed to make sure that axis 1 (or more generally: the prediction axis) is being aggregated over. Anything else doesn't work for metrics using the seasonal error because we there only is one seasonal error per data entry.

Reply:
One issue here is that you can have `TC` vs `CT` ordering.

I.e. do you store n values for each time step, or do you have n time series?

We should trust that the passed axis is the right one.

Reply:
I see, so we don't want to enforce one interpretation of the data. That makes sense.

But then, I don't know how to deal with the seasonal error. The issue here is that it's only one value per time series, so `data["seasonal_error"]` will have one dimension less than `data["label"]`. Because the aggregation (Mean for MASE and MSIS) is only taken at the end, we have to apply the seasonal error on the raw data (which is 2d in the univariate case). So, how can we know how to expand the seasonal error values?
I guess, we could just trust that the passed axis is correct, as you said, and expand along that dimension. That might work.

Reply:
Something like this should work:

```py
data = np.arange(5 * 10).reshape(5, 10)
seasonal_error = np.expand_dims(np.arange(5) + 1, axis=1)
data / seasonal_error
```

Meaning we can aggregate by an axis, and then use expand dims on the same axis on that result.

Reply:
Thanks for your suggestions. I implemented the first option, using `expand_dims`. New problem: It doesn't work with `axis=None`. In this case, I don't have an idea how to tell the seasonal error what axis to expand.

What exactly is your idea with `keep_dims`?

Reply:
As the name suggests, keep_dims will keep the dimensions. Thus, there won't be a need to expand the dims, since none were removed.

Reply:
Got it, that's a good idea! This seems to work fine.

Now there are no errors using MSIS or MASE any axis, so using the metric requires some care by the user. But I too find this better than dictating what to do with each axis.